### PR TITLE
[HotFix] Separate export chass route

### PR DESCRIPTION
**Pull request by michellemtchai**
Fix for downloading JSON has been found with original route
---------

- as described in title
- the 2 routes are:
  - `POST /export/chass/:round_id`
    - return status 200 with a message if success
    - returns status 404 with a message if failed
  - `GET /download/chass/:round_id` 
    - this path assumes that you will only download this file once. The file is deleted immediately after it's sent. You have to call the first route again to make this route work.
- fixed specs accordingly
- below is a diagram of how the routes should work
img_20170811_121609

### CODE REVIEW

**Comment by simpslandyy** [comment]
@michellemtchai all PRs should be pushed to 'develop'. Please read the message in slack regarding the development flow.
I've changed the branch this PR now points to.

**Comment by michellemtchai** [comment]
@simpslandyy Okay. I changed the target branch, but I don't see the message. Is it pinned?

**Comment by gabriellesc** [comment]
I think that this needs to be rebased off of master - there are a lot of extra commits in here

**Comment by michellemtchai** [comment]
@gabriellesc It was based off of master, the extra commits came from the `develop` when I changed the target branch for whatever strange reason. And actually, it looks like the target branch `develop` is actually out of date...

**Comment by gabriellesc** [comment]
Oh, I see! Maybe you need to rebase it off of develop, then?

Le jeu. 10 août 2017 à 23:01, michellemtchai <notifications@github.com> a
écrit :

**Comment by simpslandyy** [comment]
@michellemtchai - yeah, you may need to rebase off develop or just cherry pick and start a new branch off development. According to @gabriellesc on #156 it seemed like Git squashed the commits (and for some reason), doesn't realize that these two branches are exactly the same ... (I wanna see if I can fix that cause that's tripping me out). So it might be worth wild to just cherry pick rather than deal with "conflicts" that probably don't even exist.

File: app/controllers/download_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class DownloadController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def chass
```
Comment (jmzaleski) [inline app/controllers/download_controller.rb:4]: add comment that this is a bit of a hack.. good idea to read issue NNN about browser download extension before fiddling around

File: app/services/chass_exporter.rb [hunk 2]
```diff
@@ -10,16 +10,22 @@ def export(round_id)
...
+          return {generated: true, msg: "Offers for #{round_id} has been generated."}
         end
       else
         return {generated: false, msg: "Error: Invalid round_id"}
       end
     end
 
     private
+    def write_file(data, round_id)
+      File.open("#{Rails.root}/db/exports/offers_#{round_id}.json", "w+") do |f|
```
Comment (jmzaleski) [inline app/services/chass_exporter.rb:23]: yes, magic name!

File: app/controllers/download_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      File.delete(file_path)
+      render_helper({generated:true, data: @data, content_type: "application/json", file: "offers_#{params[:round_id]}.json"})
+    end
+  end
+
+  private
+  def render_helper(response)
+    if response[:generated]
+      send_data response[:data], filename: response[:file],
+      content_type: response[:type]
```
Comment (simpslandyy) [inline app/controllers/download_controller.rb:19]: shouldn't it be `type: response[:content_type]`?

**Pull request closed without merging**